File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/Car.swift

public class Car: Vehicle {
    
    private let vehicleType = "Car"
    private var trunkSize: Double
    private var gear: String
    private var doorsNumber: Int

    override init() {
        trunkSize = 0.0
        gear = ""
        doorsNumber = 0
        super.init()
    }
    
    init(pMake: String, pModel: String, pPlate: String, pTrunkSize: Double, pGear: String, pDoorsNumber: Int) {
        trunkSize = pTrunkSize
        gear = pGear
        doorsNumber = pDoorsNumber
        super.init(pMake: pMake, pModel: pModel, pPlate: pPlate)
    }
    
    override func getVehicleType() -> String {
        return vehicleType
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Vehicle Type: \(vehicleType)\n  -Trunk Size: \(trunkSize)\n  -Gear: \(gear)\n  -DoorsNumber: \(doorsNumber)\n"
    }
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/vehicle/Car.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.vehicle;

/**
 *
 * @author macstudent
 */
public class Car extends Vehicle {
    
    private static final String VEHICLE_TYPE = "Car";
    
    private double trunkSize;
    private String gear;
    private int doorsNumber;

    public Car(String make, String plate, double trunkSize, String gear, int doorsNumber) {
        super(make, plate);
        this.trunkSize = trunkSize;
        this.gear = gear;
        this.doorsNumber = doorsNumber;
    }

    @Override
    public String getVehicleType() {
        return VEHICLE_TYPE;
    }
}
